src/app/static/src/tests/components/userHeader.test.ts

```diff
@@ -108,4 +124,108 @@ describe("user header", () => {
         reader.readAsText(actualBlob);
     });
 
+    it ("opens file dialog on click load", (done) => {
+        const wrapper = shallowMount(UserHeader,
+            {
+                store: createStore()
+            });
+
+        wrapper.find(".dropdown-toggle").trigger("click");
+        expect(wrapper.find(".dropdown-menu").classes()).toStrictEqual(["dropdown-menu", "show"]);
+        const link = wrapper.findAll(".dropdown-item").at(1);
+
+        const input = wrapper.find("input").element as HTMLInputElement;
+        input.addEventListener("click", function(){
+            //file dialog was opened
+            done();
+        });
+
+        link.trigger("mousedown");
+    });
+
+    it("invokes load action when file selected from dialog", () => {
+        const mockLoadAction = jest.fn();
+        const wrapper = shallowMount(UserHeader,
+            {
+                store: new Vuex.Store({
+                    modules: {
+                        baseline: {
+                            namespaced: true,
+                            state: mockBaselineState()
+                        },
+                        surveyAndProgram: {
+                            namespaced: true,
+                            state: mockSurveyAndProgramState()
+                        },
+                        modelRun: {
+                            namespaced: true,
+                            state: mockModelRunState()
+                        },
+                        load: {
+                            namespaced: true,
+                            state: mockLoadState(),
+                            actions: {
+                                load: mockLoadAction
+                            }
+                        }
+                    }
+                })
+            });
+
+        const input = wrapper.find("input");
+
+        const vm = wrapper.vm;
+        const testFile = new File(["test file contents"], "filename");
+        //Can't programmatically construct a FileList to give to the read rendered input element, so we need to trick
+        //the component with a mocked ref
+        (vm.$refs as any).loadFile = {
```

Comment:
Ooh this is clever! I wonder if we can use this in the FileUpload tests to fix this: https://github.com/mrc-ide/hint/blob/master/src/app/static/src/tests/components/fileUpload.test.ts#L99